sqlmesh/core/snapshot/evaluator.py

```diff
@@ -667,6 +667,7 @@ def _create_snapshot(
                 and snapshot.is_materialized
                 and snapshot.previous_versions
                 and self.adapter.SUPPORTS_CLONING
+                and evaluation_strategy.allow_clone_for_dev_preview(snapshot)
```

Comment:
I'm a little concerned that we're adding this abstraction too soon. For example, do we definitely know that dynamic / managed tables can be cloned universally across all engines? If some engines do support this, then this should be a property of the adapter and not the strategy.

Should we not generalize prematurely and just check the model type here?

Reply:
The attempted clone here is just for creating the "dev preview" table which gets thrown away on deployment to prod. We deliberately use normal tables instead of dynamic tables for dev preview tables, so I couldnt see a situation where we would ever need to clone the production dynamic table (even if the underlying engine supports it).

The reason why I added a method on the evaluation strategy  instead of checking the model type directly within the snapshot evaluator was because I thought the whole point of evaluation strategies was to contain these checks so that the model types didn't have to leak into the snapshot evaluator.

Would it be better to adjust the check to:
```
and snapshot.model.kind != ManagedKind
```
?

Reply:
well the fact that we have `snapshot.is_materialized` in there means that this leakage already occurred :) So if anything adding `snapshot.is_managed` there will make it consistent.

Reply:
Ok, i've updated the check to use the `is_managed` property (which I forgot existed) and added a comment explaining why